File: pdu/python/hako_msgs/pdu_cdr_conv_HakoCmdHeader.py
from .pdu_pytype_HakoCmdHeader import HakoCmdHeader
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_HakoCmdHeader(writer: CdrWriter, src: HakoCmdHeader):
    writer.write_bool(src.request)
    writer.write_bool(src.result)
    writer.write_int32(src.result_code)


def cdr_body_to_py_HakoCmdHeader(reader: CdrReader, dst: HakoCmdHeader):
    dst.request = reader.read_bool()
    dst.result = reader.read_bool()
    dst.result_code = reader.read_int32()
    return dst


def py_to_cdr_HakoCmdHeader(src: HakoCmdHeader) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_HakoCmdHeader(writer, src)
    return writer.bytes()


def cdr_to_py_HakoCmdHeader(cdr_payload) -> HakoCmdHeader:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = HakoCmdHeader()
    return cdr_body_to_py_HakoCmdHeader(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_HakoCmdHeader.js
import { HakoCmdHeader } from './pdu_jstype_HakoCmdHeader.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduHakoCmdHeaderConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { HakoCmdHeader } src
     */
    static to_cdr_body(writer, src) {
        writer.write_bool(src.request);
        writer.write_bool(src.result);
        writer.write_int32(src.result_code);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { HakoCmdHeader } dst
     * @returns { HakoCmdHeader }
     */
    static cdr_body_to_js(reader, dst) {
        dst.request = reader.read_bool();
        dst.result = reader.read_bool();
        dst.result_code = reader.read_int32();
        return dst;
    }

    /**
     * @param { HakoCmdHeader } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { HakoCmdHeader }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new HakoCmdHeader());
    }
}
